### ISSUE TITLE
New Message Format 

### ISSUE THREAD

**Issue by c-leitner** [open918/open918#1]
QR-Codes with the message format v2 have appeared in Austria (First chars of ticket: #UT02118100301O). No library currently supports the message format and I think it has something to do with how the data is being compressed. Zlib is not able to decompress the payload with message format v2. I am currently trying to get my hands on the update standard for electronic tickets UIC-90918-8 (https://www.shop-etf.com/en/layout-for-electronically-issued-rail-passenger-tickets) but the price is a little bit steep. If I find the problem I will submit a pull request.

**Comment by joelhaasnoot** [comment]
Do you have a sample code by any chance?

**Comment by c-leitner** [comment]
Yes I do, but it has my personal information in it. I could send it to you via E-Mail. Unfortunately I think it is close to impossible to decode it with Java because the data is encode with "ASN1 PER unaligned" and I could not find a compiler that reads that format in Java. The XML File and the documentation of the message format is fortunately available at ERA. I also managed to decompress the data with zlib but it was encoded in ASN1. This is why I thought zlib could not decompress it.
Ticket format documentation:
https://www.era.europa.eu/sites/default/files/library/docs/recommendation/era_rec122_tap_tsi_revision_recommendation_technical_document_b12_en.pdf
XML Files:
https://www.era.europa.eu/sites/default/files/xml_schema_files.zip
Here is also a Barcode formatted in ASN but I am unsure about its exact contents:
https://www.era.europa.eu/sites/default/files/barcode.zip
All the links were taken from here https://www.era.europa.eu/library/era-recommendations_en belonging to "Recommendation ERA-REC-122 of the European Union Agency for Railways on the technical specification for Interoperability relating to the subsystem 'telematics applications for passenger services' of the Union rail system"

**Comment by joelhaasnoot** [comment]
We accept submissions of samples at barcode@bliksemlabs.com . The ticket sounds like something I've been able to parse before, just not in this branch (long story...).

**Comment by craftbyte** [comment]
If anyone is still looking into this issue/repo, I would suggest you try checking out the official UIC library that supports all the currently used standards (barring SSBv1 used for domestic tickets in Finland) at https://github.com/UnionInternationalCheminsdeFer/UIC-barcode. There was also a talk at 38c3 regarding this: https://youtu.be/N17GQJBIaiU

**Comment by joelhaasnoot** [comment]
Great, thanks for sharing. Will take a look! The 38C3 video is indeed in my watch list. 

One goal for this library that it looks like the UIC lib doesn't tackle is mapping the older tickets to logical fields (i.e. what is the destination), something it looks like the new tickets have better defined/standardized. In other words - it would be nice to have a higher level API that doesn't care about which ticket type it is but can return fields
